src/setups/rectangular_shape.jl

```diff
@@ -89,6 +92,13 @@ function RectangularShape(particle_spacing, n_particles_per_dimension, min_coord
                                            min_coordinates, tlsph=tlsph,
                                            loop_order=loop_order)
 
+    if !isnothing(coordinates_perturbation)
+        seed!(1)
+        amplitude = coordinates_perturbation * particle_spacing
+        coordinates .+= rand((-amplitude):(particle_spacing * 1e-3):(amplitude),
+                             NDIMS, n_particles)
+    end
```

Comment:
If you want to do this here, can you please add the full `rectangular_patch` machinery from the test directory?

Reply:
I was first also thinking about that. But we only need a random variation of the coordinates and not the quantities. I can't think of a use case where one would like to initially perturb the quantities of particles but except in tests. 
`rectangular_patch`  returns `InitialConditon` and we only need perturbed coordinates.

Reply:
`rectangular_patch` has two perturbation parameters. Coordinate perturbation factor and quantity perturbation factor. You could just set the latter to 0 by default.

Reply:
I created an Issue (#620). I think it's better to put this in a new PR